llmfoundry/data/finetuning/tasks.py

```diff
@@ -365,6 +367,15 @@ def build_from_hf(
             with dist.local_rank_zero_download_and_wait(signal_file_path):
                 pass
 
+        hf_tokenization_logger = logging.getLogger(
+            'transformers.tokenization_utils_base')
+        sequence_length_warning_filter = SpecificWarningFilter(
+            'Token indices sequence length is longer than the specified maximum sequence length'
+        )
+
+        # We will trim examples later in the collate_fn, so we want to silence this warning from Hugging Face
+        hf_tokenization_logger.addFilter(sequence_length_warning_filter)
```

Comment:
I think it could be nice to have a context manager helper for cases where the filter should be removed as well

thinking something roughly like:
```
# same file as SpecificWarningFilter definition
from contextlib import contextmanager

@contextmanager
def filtered_logging(logger, filtered_msg):
    filter = SpecificWarningFilter(filtered_msg)
    try:
        logger.addFiler(filter)
    finally:
        logger.removeFilter(filter)
        
```

Usage:
```
with filtered_logging(hf_tokeniznization_logger, 'Token indices sequence length is longer than the specified maximum sequence length'):
    ...

```

Reply:
I don't really love adding the extra level of indentation require to use the context manager, and since this only occurs in one place right now, I think I will leave it. Happy to revisit if it starts popping up more though.

Reply:
maybe this is better/more consistent naming `with filtered_logging(...)` => `with specific_warning_filter(...)`